src/AppInstallerCLIE2ETests/Interop/InstallInterop.cs

```diff
@@ -516,5 +517,51 @@ public async Task InstallRequireUserScopeAndUnknown()
             // Assert
             Assert.AreEqual(InstallResultStatus.Ok, installResult.Status);
         }
+
+        /// <summary>
+        /// Test installing package with agreements and accepting those agreements.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Test]
+        public async Task InstallWithAgreementsAccepted()
+        {
+            // Find package
+            var searchResult = this.FindOnePackage(this.testSource, PackageMatchField.Id, PackageFieldMatchOption.Equals, "AppInstallerTest.CatalogPackageMetadata");
+
+            // Configure installation
+            var installOptions = this.TestFactory.CreateInstallOptions();
+            installOptions.PackageInstallMode = PackageInstallMode.Silent;
+            installOptions.PreferredInstallLocation = this.installDir;
+            installOptions.AcceptPackageAgreements = true;
+
+            // Install
+            var installResult = await this.packageManager.InstallPackageAsync(searchResult.CatalogPackage, installOptions);
+
+            // Assert
+            Assert.AreEqual(InstallResultStatus.Ok, installResult.Status);
+        }
+
+        /// <summary>
+        /// Test installing package with agreements and not accepting those agreements.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Test]
+        public async Task InstallWithAgreementsNotAccepted()
+        {
+            // Find package
+            var searchResult = this.FindOnePackage(this.testSource, PackageMatchField.Id, PackageFieldMatchOption.Equals, "AppInstallerTest.CatalogPackageMetadata");
+
+            // Configure installation
+            var installOptions = this.TestFactory.CreateInstallOptions();
+            installOptions.PackageInstallMode = PackageInstallMode.Silent;
+            installOptions.PreferredInstallLocation = this.installDir;
+            installOptions.AcceptPackageAgreements = false;
+
+            // Install
+            var installResult = await this.packageManager.InstallPackageAsync(searchResult.CatalogPackage, installOptions);
+
+            // Assert
+            Assert.AreEqual(InstallResultStatus.CatalogError, installResult.Status);

```

Comment:
>Assert.AreEqual(InstallResultStatus.CatalogError, installResult.Status); [](http://example.com/codeflow?start=12&length=72)

I don't think CatalogError should be returned if agreements were not accepted

Reply:
Changed to new error - 'PackageAgreementsNotAccepted'